backend/products/models.py

```diff
@@ -0,0 +1,59 @@
+from django.db import models
+
+# Create your models here.
+
+class Product(models.Model):
+    name = models.CharField(max_length=200)
+    description = models.TextField(max_length=600, blank=True)
+    category = models.CharField(max_length=100)
+    primary_variant = models.OneToOneField(
+        'ProductVariation', on_delete=models.CASCADE, default=None, null=True)
+    
+    # Need to add Primary Variant and seller Id
+    created_at = models.DateTimeField(auto_now_add=True)
+    updated_at = models.DateTimeField(auto_now=True)
+
+    def __str__(self):
+        return self.name
+    
+
+class ProductVariation(models.Model):
+    product_id = models.ForeignKey(Product, on_delete=models.CASCADE)
+    color_id = models.ForeignKey('ProductColor', on_delete=models.CASCADE)
+    size_id = models.ForeignKey('ProductSize', on_delete=models.CASCADE)
+    price = models.IntegerField()
+    quantity = models.IntegerField()
+    cover_image = models.ForeignKey('ProductImage', on_delete=models.CASCADE, default=None, null=True)
+    is_active = models.BooleanField(default=True)
+
+    required = ['product_id', 'color_id', 'size_id', 'price', 'quantity']
```

Comment:
does this make the fields required? I'm not familiar with this method could you link a reference?
Also all fields are required by default in django
[reference](https://docs.djangoproject.com/en/4.2/topics/db/models/#:~:text=If%20True%2C%20the,will%20be%20required.)